reports/models.py

```diff
@@ -0,0 +1,402 @@
+    # keep some information so the data is still available for reports:
+    #     1. Group sizes: Kids and adults
+    #     2. Study levels
+    # - Store the unit position (coordinates) and divisions to normalise the data.
+    # Place data must be fetched from LinkedEvents API
+    # (or from other service like the servicemap API)
+
+    _study_group = models.OneToOneField(
+        occurrences_models.StudyGroup,
+        on_delete=models.SET_NULL,
+        null=True,
+        db_column="study_group_id",
+    )
+    study_group_unit_id = models.CharField(
+        max_length=255, verbose_name=_("study group's unit id"), null=True
+    )
+    study_group_amount_of_children = models.PositiveSmallIntegerField(
+        verbose_name=_("amount of the children in the study group")
+    )
+    study_group_amount_of_adult = models.PositiveSmallIntegerField(
+        verbose_name=_("amount of the adults in the study group"), default=0
+    )
+    study_group_study_levels = ArrayField(
+        # ArrayField Size 2: Study level id, translated name
+        ArrayField(models.CharField(max_length=255), size=2),
+        verbose_name=_("study levels of the study group"),
+    )
+    # The position of the (unit) place from LinkedEvents API
+    study_group_unit_position = ArrayField(
+        models.DecimalField(max_digits=9, decimal_places=6),
+        size=2,
+        null=True,
+        verbose_name=_("coordinates of the study group unit"),
+    )
+    # The divisions of the (unit) place from LinkedEvents API
+    study_group_unit_divisions = ArrayField(
+        models.CharField(max_length=200),
+        verbose_name=_("study group unit divisions"),
+        help_text=_("Open Civic Data identifiers"),
+        null=True,
+    )
+
+    # Enrolment
+    # - Store the foreign key to fetch all data
+    # - If the foreign key integrity is gone,
+    # keep some information so the data is still available for reports:
+    #     1. enrolment time
+    #     2. enrolment status
+    # - Enrolment status is also used to check whether or not
+    # the db row should be updated.
+    _enrolment = models.OneToOneField(
+        occurrences_models.Enrolment,
+        on_delete=models.SET_NULL,
+        null=True,
+        db_column="enrolment_id",
+    )
+    enrolment_time = models.DateTimeField(verbose_name=_("enrolment time"))
+    enrolment_status = models.CharField(
+        max_length=255, verbose_name=_("enrolment status")
+    )
+
+    # Occurrence
+    # - Store the foreign key to fetch all data
+    # - If the foreign key integrity is gone,
+    # keep some information so the data is still available for reports:
+    #     1. Amount of the seats
+    #     2. cancelled status
+    # - Store the occurrence languages
+    # - Store the event position (coordinates) and divisions to normalise the data.
+    # Place data must be fetched from LinkedEvents API
+    # (or from other service like the servicemap API)
+    _occurrence = models.OneToOneField(
```

Comment:
I think only `_enrolment` can be `OneToOneField` since a single occurrence can have multiple enrolments? Same probably applies to the `_study_group` as well.

Reply:
Changed them to be onetoones